octavia-cli/octavia_cli/apply/resources.py

```diff
@@ -595,8 +599,20 @@ def _deserialize_raw_configuration(self):
         self._check_for_legacy_connection_configuration_keys(configuration)
         configuration["sync_catalog"] = self._create_configured_catalog(configuration["sync_catalog"])
         configuration["namespace_definition"] = NamespaceDefinitionType(configuration["namespace_definition"])
-        if "schedule" in configuration:
-            configuration["schedule"] = ConnectionSchedule(**configuration["schedule"])
+
+        if "schedule_type" in configuration:
+            # If schedule type is manual we do not expect a schedule_data field to be set
+            # TODO: sending a WebConnectionCreate payload without schedule_data (for manual) fails.
```

Comment:
@mfsiega-airbyte I tried to send a payload with only `schedule_type` set to `manual` but received an error response saying `schedule_data` must be set. Could you please double-check the behavior of this function and if it matches the expected logic on the API side? I'd be thankful if you could also update the corresponding unit test (`test__deserialize_raw_configuration`) to ensure all cases are properly handled.

Reply:
Turns out this was happening when updating from manual -> non-manual; fixed this in the API.

Added a bit of test coverage as well.